### PR TITLE
:bug: fix: ItemLikeRepository Query refactoring to queryDSL

### PR DESCRIPTION
**Pull request by seonghun-dev**
- Resolve : https://github.com/depromeet/street-drop-server/issues/392
### Before
```
Hibernate: select i1_0.item_like_id,i1_0.created_at,i2_0.item_id,a1_0.album_cover_id,a1_0.album_image,a1_0.album_thumbnail,i2_0.content,i2_0.created_at,i3_0.item_location_id,i3_0.created_at,i3_0.item_id,i3_0.modified_at,i3_0.name,i3_0.point,i3_0.village_id,i2_0.modified_at,s1_0.song_id,a2_0.album_id,a2_0.album_cover_id,a2_0.artist_id,a2_0.name,s1_0.name,i2_0.user_id,i1_0.modified_at,i1_0.user_id from item_like i1_0 join item i2_0 on i2_0.item_id=i1_0.item_id join item_location i3_0 on i2_0.item_id=i3_0.item_id join album_cover a1_0 on a1_0.album_cover_id=i2_0.album_cover_id join song s1_0 on s1_0.song_id=i2_0.song_id join album a2_0 on a2_0.album_id=s1_0.album_id where i1_0.user_id=? and i1_0.item_like_id<? order by i1_0.item_like_id desc
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select g1_0.genre_id,g1_0.name from genre g1_0 where g1_0.genre_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select u1_0.user_id,u1_0.created_at,u1_0.idfv,u1_0.modified_at,u1_0.music_app,u1_0.nickname,u1_0.user_level_id from users u1_0 where u1_0.user_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
Hibernate: select a1_0.artist_id,a1_0.name from artist a1_0 where a1_0.artist_id=?
Hibernate: select g1_0.song_id,g1_0.song_genre_id,g1_0.genre_id from song_genre g1_0 where g1_0.song_id=?
Hibernate: select l1_0.item_id,l1_0.item_like_id,l1_0.created_at,l1_0.modified_at,l1_0.user_id from item_like l1_0 where l1_0.item_id=?
```

### After (SubQuery)

SubQuery SQL
``` sql
select abs(weekofyear(il.created_at) - weekofyear(current_date)) as weekAgo, 
i.item_id, i.content, i.created_at, 
i_loc.name as locationName,
s.name as songName, 
alb.name as albumName, 
art.name as artistName,
alb_c.album_thumbnail, 
i.user_id as userId,
u.nickname as userNickname,
(select count(il2.item_like_id) from item_like il2 where il2.item_id = i.item_id) as ItemLikeCount
from item_like il
join item i on i.item_id= il.item_id
join item_location i_loc on i.item_id=i_loc.item_id 
join song s on i.song_id=s.song_id 
join album alb on s.album_id=alb.album_id 
join artist art on alb.artist_id=art.artist_id 
join album_cover alb_c on i.album_cover_id=alb_c.album_cover_id 
join users u on u.user_id=i.user_id 
where il.user_id = ?
order by il.item_like_id desc;
```

SubQuery Explain

id | select_type | table | partitions | type | possible_keys | key | key_len | ref | rows | filtered | Extra
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | PRIMARY | il | NULL | ref | asdf,user_index_adf | user_index_adf | 9 | const | 13 | 100.00 | Using where; Backward index scan
1 | PRIMARY | i | NULL | eq_ref | PRIMARY,FKip7vjlqnw0rt125paig31ni3,FKf9j8ocdanjs7a48djb3aqd0en,FKc4s174l330le17rblwgyjawev | PRIMARY | 8 | streetdrop.il.item_id | 1 | 100.00 | Using where
1 | PRIMARY | i_loc | NULL | ref | FK50oxdirrbr9uk55y7xlrq6jmv | FK50oxdirrbr9uk55y7xlrq6jmv | 8 | streetdrop.il.item_id | 1 | 100.00 | NULL
1 | PRIMARY | u | NULL | eq_ref | PRIMARY | PRIMARY | 8 | streetdrop.i.user_id | 1 | 100.00 | NULL
1 | PRIMARY | s | NULL | eq_ref | PRIMARY,FKrcjmk41yqj3pl3iyii40niab0 | PRIMARY | 8 | streetdrop.i.song_id | 1 | 100.00 | NULL
1 | PRIMARY | alb | NULL | eq_ref | PRIMARY,FKmwc4fyyxb6tfi0qba26gcf8s1 | PRIMARY | 8 | streetdrop.s.album_id | 1 | 100.00 | NULL
1 | PRIMARY | art | NULL | eq_ref | PRIMARY | PRIMARY | 8 | streetdrop.alb.artist_id | 1 | 100.00 | NULL
1 | PRIMARY | alb_c | NULL | eq_ref | PRIMARY | PRIMARY | 8 | streetdrop.i.album_cover_id | 1 | 100.00 | NULL
2 | DEPENDENT SUBQUERY | il2 | NULL | ref | asdf | asdf | 9 | streetdrop.i.item_id | 1 | 100.00 | Using index

### After(Lateral Join)

Lateral SQL
``` sql
select abs(weekofyear(il.created_at) - weekofyear(current_date)) as weekAgo, 
i.item_id, i.content, il.created_at, 
i_loc.name as locationName,
s.name as songName, 
alb.name as albumName, 
art.name as artistName,
alb_c.album_thumbnail, 
cnt_v.cnt as itemLikeCount,
i.user_id as userId,
u.nickname as userNickname
from item_like il
join item i on i.item_id=il.item_id 
join item_location i_loc on i.item_id=i_loc.item_id 
join song s on i.song_id=s.song_id 
join album alb on s.album_id=alb.album_id 
join artist art on alb.artist_id=art.artist_id 
join album_cover alb_c on i.album_cover_id=alb_c.album_cover_id 
join users u on u.user_id=i.user_id 
inner join lateral (select count(il2.item_like_id) as cnt from item_like il2 where il2.item_id = i.item_id) cnt_v on true
where il.user_id = ?
order by il.item_like_id desc;
```

Lateral Explain Result
id | select_type | table | partitions | type | possible_keys | key | key_len | ref | rows | filtered | Extra
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | PRIMARY | il | NULL | ref | asdf,user_index_adf | user_index_adf | 9 | const | 13 | 100.00 | Using where; Backward index scan
1 | PRIMARY | i | NULL | eq_ref | PRIMARY,FKip7vjlqnw0rt125paig31ni3,FKf9j8ocdanjs7a48djb3aqd0en,FKc4s174l330le17rblwgyjawev | PRIMARY | 8 | streetdrop.il.item_id | 1 | 100.00 | Rematerialize (<derived2>)
1 | PRIMARY | i_loc | NULL | ref | FK50oxdirrbr9uk55y7xlrq6jmv | FK50oxdirrbr9uk55y7xlrq6jmv | 8 | streetdrop.il.item_id | 1 | 100.00 | NULL
1 | PRIMARY | u | NULL | eq_ref | PRIMARY | PRIMARY | 8 | streetdrop.i.user_id | 1 | 100.00 | NULL
1 | PRIMARY | s | NULL | eq_ref | PRIMARY,FKrcjmk41yqj3pl3iyii40niab0 | PRIMARY | 8 | streetdrop.i.song_id | 1 | 100.00 | NULL
1 | PRIMARY | alb | NULL | eq_ref | PRIMARY,FKmwc4fyyxb6tfi0qba26gcf8s1 | PRIMARY | 8 | streetdrop.s.album_id | 1 | 100.00 | NULL
1 | PRIMARY | art | NULL | eq_ref | PRIMARY | PRIMARY | 8 | streetdrop.alb.artist_id | 1 | 100.00 | NULL
1 | PRIMARY | alb_c | NULL | eq_ref | PRIMARY | PRIMARY | 8 | streetdrop.i.album_cover_id | 1 | 100.00 | NULL
1 | PRIMARY | <derived2> | NULL | ALL | NULL | NULL | NULL | NULL | 2 | 100.00 | NULL
2 | DEPENDENT DERIVED | il2 | NULL | ref | asdf | asdf | 9 | streetdrop.i.item_id | 1 | 100.00 | Using index

### CODE REVIEW

File: backend/streetdrop-api/src/main/java/com/depromeet/domains/user/service/UserItemService.java [hunk 1]
```diff
@@ -99,40 +87,7 @@ private ItemGroupResponseDto itemDaotoItemGroupResponseDto(User user, ItemDao it
...
-                        )
-                        .entrySet()
-                        .stream()
-                        .map(entry -> new ItemGroupByDateResponseDto(entry.getKey(), entry.getValue()))
-                        .sorted(Comparator.comparingInt(dto -> getWeekString2Int(dto.date())))
-                        .toList();
-        var meta = new InfiniteScrollMetaResponseDto(itemLikeList.size(), -1);
-
-        return new InfiniteScrollResponseDto<>(itemGroupByDateResponseDtoList, meta);
+        return itemLikeService.getLikedItemsByUser(user, nextCursor);
```
Comment (seonghun-dev) [inline backend/streetdrop-api/src/main/java/com/depromeet/domains/user/service/UserItemService.java:90]: User 도메인에서는 Item관련 서비스만 의존하도록 하였습니다

File: backend/streetdrop-api/src/main/java/com/depromeet/domains/item/service/ItemLikeService.java [hunk 2]
```diff
@@ -42,4 +54,47 @@ public void unlikeItem(User user, Long itemId) {
...
+					return new ItemGroupByDateResponseDto(getWeeksAgo(value), itemGroupResponseDtoList);
+				})
+				.toList();
+
+		var meta = new InfiniteScrollMetaResponseDto(itemLikeDaoList.size(), -1);
+
+		return new InfiniteScrollResponseDto<>(itemGroupByDateResponseDto, meta);
+	}
+
+	private ItemGroupResponseDto userItemLikeDaotoItemGroupResponseDto(UserItemLikeDao userItemLikeDao) {
```
Comment (seonghun-dev) [inline backend/streetdrop-api/src/main/java/com/depromeet/domains/item/service/ItemLikeService.java:81]: 추후 Mapper 통합하는 추가 이슈를 진행하겠습니다

File: backend/streetdrop-api/src/main/java/com/depromeet/domains/user/service/UserItemService.java (hunk 1 above)
Comment (siyeonSon) [reply backend/streetdrop-api/src/main/java/com/depromeet/domains/user/service/UserItemService.java:90]: UserItemService가 ItemLikeService만 바라보게 변경하면 좋겠군요!

아래의 `getLikedItemsPoints()` 쪽도 ItemLikeRepository를 의존하지 않고, ItemLikeService를 의존하도록 수정해 놓겠습니다~

Comment (seonghun-dev) [reply backend/streetdrop-api/src/main/java/com/depromeet/domains/user/service/UserItemService.java:90]: 나중에 다른 유저 정보 조회할때 구현을 간편하게 할 수 있을 것 같아요

Comment (seonghun-dev) [reply backend/streetdrop-api/src/main/java/com/depromeet/domains/user/service/UserItemService.java:90]: 나중에 다른 유저 정보 조회할때 구현을 간편하게 할 수 있을 것 같아요

**Review by siyeonSon — APPROVED** [review]

**Pull request merged**